### ISSUE TITLE
Generated Razor components

### ISSUE THREAD

**Issue by timeshift92** [ChilliCream/hotchocolate#4476]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

![image](https://user-images.githubusercontent.com/13614530/143259030-81578954-9986-4bad-8e1d-fe73b366f4ff.png)

### Steps to reproduce

1. run any hello world with a generated component, it will give an error

### Relevant log output

```shell
System.InvalidOperationException
  HResult=0x80131509
  Message=The current thread is not associated with the Dispatcher. Use InvokeAsync() to switch execution to the Dispatcher when triggering rendering or component state.
  Source=Microsoft.AspNetCore.Components
  StackTrace:
   at Microsoft.AspNetCore.Components.Dispatcher.AssertAccess() in /_/src/Components/Components/src/Dispatcher.cs:line 42
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.AddToRenderQueue(Int32 componentId, RenderFragment renderFragment) in /_/src/Components/Components/src/RenderTree/Renderer.cs:line 513
   at Microsoft.AspNetCore.Components.ComponentBase.StateHasChanged() in /_/src/Components/Components/src/ComponentBase.cs:line 111
   at StrawberryShake.Razor.QueryBase`1.<Subscribe>b__23_0(IOperationResult`1 operationResult) in /home/vsts/work/1/s/src/StrawberryShake/Client/src/Razor/QueryBase.cs:line 45
   at System.Reactive.AnonymousObserver`1.OnNextCore(T value) in /_/Rx.NET/Source/src/System.Reactive/AnonymousObserver.cs:line 67
   at System.Reactive.ObserverBase`1.OnNext(T value) in /_/Rx.NET/Source/src/System.Reactive/ObserverBase.cs:line 36
   at StrawberryShake.OperationExecutor`2.OperationExecutorObservable.<ExecuteAsync>d__8.MoveNext() in /home/vsts/work/1/s/src/StrawberryShake/Client/src/Core/OperationExecutor.Observable.cs:line 94
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/ExceptionServices/ExceptionDispatchInfo.cs:line 56
   at StrawberryShake.OperationExecutor`2.OperationExecutorObservable.<ExecuteAsync>d__8.MoveNext() in /home/vsts/work/1/s/src/StrawberryShake/Client/src/Core/OperationExecutor.Observable.cs:line 79

  This exception was originally thrown at this call stack:
    Microsoft.AspNetCore.Components.Dispatcher.AssertAccess() in Dispatcher.cs
    Microsoft.AspNetCore.Components.RenderTree.Renderer.AddToRenderQueue(int, Microsoft.AspNetCore.Components.RenderFragment) in Renderer.cs
    Microsoft.AspNetCore.Components.ComponentBase.StateHasChanged() in ComponentBase.cs
    StrawberryShake.Razor.QueryBase<TResult>.Subscribe.AnonymousMethod__23_0(StrawberryShake.IOperationResult<TResult>) in QueryBase.cs
    System.Reactive.AnonymousObserver<T>.OnNextCore(T) in AnonymousObserver.cs
    System.Reactive.ObserverBase<T>.OnNext(T) in ObserverBase.cs
    StrawberryShake.OperationExecutor<TData, TResult>.OperationExecutorObservable.ExecuteAsync(System.IObserver<StrawberryShake.IOperationResult<TResult>>, StrawberryShake.ObserverSession) in OperationExecutor.Observable.cs
    System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in ExceptionDispatchInfo.cs
    StrawberryShake.OperationExecutor<TData, TResult>.OperationExecutorObservable.ExecuteAsync(System.IObserver<StrawberryShake.IOperationResult<TResult>>, StrawberryShake.ObserverSession) in OperationExecutor.Observable.cs
```

### Product

Strawberry Shake

### Version

12.3.1

**Comment by michaelstaib** [comment]
Is that with Blazor for server or Blazor Wasm?

**Comment by timeshift92** [comment]
it is that with Blazor for server

**Comment by PascalSenn** [comment]
Closes in favour of #4430
